GUI/Types/Renderer/Shaders/post_processing.frag

```diff
@@ -0,0 +1,125 @@
+#version 460
+
+#include "common/utils.glsl"
+
+//#define F_OUTPUT_LINEAR 0
+#define F_COLOR_CORRECTION_LUT 0
+#define F_HAS_BLUE_NOISE 0
+#define F_IS_POST_HLA 0
+//#define F_BLOOM 0
+//#define F_OLD_TONEMAPPING_CURVE 0 // HLA, linear only. Used in linear viewmodes to disable the tonemapping curve
+
+// Bloom might eventually be supported
+//uniform sampler2D g_tBloom; // t13
+//uniform vec4 g_vBloomUvScaleClamp;
+//uniform vec3 g_vNormalizedBloomStrengths;
+//uniform vec3 g_vUnNormalizedBloomStrengths;
+
+
+
+
+layout(location=0) uniform sampler2DMS g_tColorBuffer;
+
+vec4 SampleColorBuffer(vec2 coords)
+{
+    vec4 singleSampleColor = texelFetch(g_tColorBuffer, ivec2(coords.xy), int(gl_SampleID));
+    singleSampleColor.rgb = singleSampleColor.rgb / (max3(singleSampleColor.rgb) + 1.0);
+    return singleSampleColor;// / 2.8;
+/*
+// This pretty much implements msaa_resolve_cs as a pixel shader, and without the 4x downsample
+    vec4 msaaResolveColor = vec4(0.0);
+    float INV_SAMPLES = 1.0 / F_MSAA_SAMPLES;
+
+    for (uint i = uint(F_MSAA_SAMPLES); i < F_MSAA_SAMPLES; i++)
+    {
+        vec4 singleSampleColor = texelFetch(g_tColorBuffer, ivec2(coords.xy), int(i));
+        singleSampleColor = clamp(singleSampleColor, vec4(0.0), vec4(65504.0));
+
+        msaaResolveColor += singleSampleColor;
+    }
+    return msaaResolveColor;
+*/
+}
+
+
+
+uniform float g_flToneMapScalarLinear;
+uniform float g_flExposureBiasScaleFactor;
+uniform float g_flShoulderStrength;
+uniform float g_flLinearStrength;
+uniform float g_flLinearAngle;
+uniform float g_flToeStrength;
+uniform float g_flToeNum;
+uniform float g_flToeDenom;
+uniform float g_flWhitePointScale;
+
+vec3 TonemapColor(vec3 vColor)
+{
+    vColor *= (g_flToneMapScalarLinear * g_flExposureBiasScaleFactor) * 2.8;
+
+    // Uncharted tonemapper
+    vec3 tonemapNumerator = vColor * (g_flShoulderStrength * vColor + g_flLinearStrength * g_flLinearAngle) + g_flToeNum * g_flToeStrength;
+    vec3 tonemapDenominator = vColor * (g_flShoulderStrength * vColor + g_flLinearStrength) + g_flToeDenom * g_flToeStrength;
+
+    vec3 vTonemappedColor = tonemapNumerator / tonemapDenominator - g_flToeNum / g_flToeDenom;
+
+    // Divide by tonemapped white point (pre-calculated on CPU)
+    vTonemappedColor *= g_flWhitePointScale; // This is actually 1/TonemapColor(WhitePoint)
+
+    // Finally, convert from Linear to Gamma space
+    vec3 vTonemappedColorSRGB = SrgbLinearToGamma(vTonemappedColor);
+
+    return vTonemappedColorSRGB;
+}
+
+
+
+#if F_COLOR_CORRECTION_LUT
```

Comment:
I think we might be able to use this shader without any define, and use uniform bools if really necessary.

Reply:
We'd have to provide a texture for g_tColorCorrection still though, right?

Reply:
With a boolean check, maybe not.

(now I'm wondering what a passthrough LUT would be like lol)